rfcs/0040-rosetta-timelocking.md

```diff
@@ -0,0 +1,115 @@
+
+- For `/account/balance` queries involving the time-locked accounts, use the time-locking functions that already exist https://github.com/MinaProtocol/mina/blob/92ea2c06523559b9980658d15b9e5271400ac856/src/lib/coda_base/account.ml#L561 using the timing info either in the genesis ledger (Approach A) or the database (Approach B) and the balance we found above.
+
+## Drawbacks
+
+[drawbacks]: #drawbacks
+
+This will increase the size of our archived data by a factor of around two. This seems acceptable.
+
+## Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+### Use a Rosetta operation to unlock tokens each block
+
+The nice thing about this approach is we wouldn't have to change our protocol, and we use operations -- the atomic unit of change in Rosetta -- to model tokens vesting.
+
+This approach is not ideal from a scalability perspective because we would need to generate synthetic operations adding the liquid balances to every single currently vesting account.
+
+There is also another more serious reason that this approach is unacceptable: Floating-point rounding issues will cause the sum of the parts to not equal the whole. In other words, summing each of the synthetic operations growing the liquid balance up until block `b`, would not be equal to querying the liquid balance at block `b` itself.
+
+### Change the protocol
+
+Other protocols that have similar time-locked accounts require an explicit on-chain transaction to move liquid funds out of the vesting account before they are actually usable. If we changed our protocol to support such a transaction, it would be trivial to model this in Rosetta.
+
+However, this provides a worse experience for users. Even though they know their account has liquid funds, and even though the _protocol_ knows their account has liquid funds, a separate transaction is required before they're usable.
+
+Additionally, we want to avoid changing the protocol this close to a looming mainnet launch.
+
+## Prior art
+
+[prior-art]: #prior-art
+
+Celo's and Solana's Rosetta implementation is similar to the "Change the protocol" section in Rationale and Alternatives.
+
+## Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+I would like at least (2) and (3) resolved before we do significant implementation work on this feature.
+
+1. Will the Rosetta team officially be okay with this approach, even in the world where we need time locked account creation after genesis (and thus need a non-static balance exemption list) . See this discourse thread for more info:
+   https://community.rosetta-api.org/t/representing-minas-vesting-accounts-using-balance-exemptions/317
+
+2. Is Approach A actually easier if we don't yet support time-locked account creation after genesis? We will eventually need to support this later, are we wasting time if we don't just go with Approach B?
```

Comment:
I would prefer having it in the table because then you have the option to calculate the liquid balance just by querying the database.
And if we eventually have to support time-locked account creation after genesis then doing it before main-net would save us an archive db migration.